sdv/constraints/tabular.py

```diff
@@ -300,24 +325,30 @@ def _fit(self, table_data):
             table_data (pandas.DataFrame):
                 The Table data.
         """
-        if self._high_is_scalar is None:
-            self._high_is_scalar = self._high not in table_data.columns
-        if self._low_is_scalar is None:
-            self._low_is_scalar = self._low not in table_data.columns
-
-        if self._high_is_scalar and self._low_is_scalar:
-            raise TypeError('`low` and `high` cannot be both scalars at the same time')
-        elif self._low_is_scalar:
-            self.constraint_columns = (self._high,)
-            self._dtype = table_data[self._high].dtype
-        elif self._high_is_scalar:
-            self.constraint_columns = (self._low,)
-            self._dtype = table_data[self._low].dtype
+        if self._scalar == 'high':
+            exists = [column in table_data.columns for column in self._low]
```

Comment:
This block and the one below could be moved to a method which is called one way or another depending on `_scalar`

Reply:
+1, we can move the duplicated logic in L329-L332 to a class method `_check_columns_exist` and call it with either `self._low` or `self._high`.